packages/marshal/src/marshal.js

```diff
@@ -998,17 +1003,16 @@ export function makeMarshal(
         }
         return ibidTable.finish(result);
       } else {
-        const result = ibidTable.start({});
+        let result = ibidTable.start({});
         const names = ownKeys(rawTree);
-        for (const name of names) {
-          assert.typeof(name, 'string');
-          result[name] = fullRevive(rawTree[name]);
-        }
         if (names.length === 0) {
-          Object.defineProperty(result, PASS_STYLE, {
-            enumerable: false,
-            value: 'copyRecord',
-          });
+          // eslint-disable-next-line no-use-before-define
+          result = Data(result);
+        } else {
+          for (const name of names) {
+            assert.typeof(name, 'string');
```

Comment:
Please add a description, such as:

```suggestion
            assert.typeof(name, 'string', X`String-named property ${name} of ${rawTree} not allowed`);
```

Reply:
Thanks. Changed the wording to
```
X`Property ${name} of ${rawTree} must be a string`
```